Realm/Tests/MigrationTests.mm

```diff
@@ -1133,21 +1133,35 @@ - (void)testEnumerateObjectsAfterDeleteDataForRemovedType {
         [IntObject createInRealm:realm withValue:@[@2]];
     }];
 
-    RLMRealmConfiguration *config = self.config;
-    config.objectClasses = @[StringObject.class];
-    config.schemaVersion = 1;
-    config.migrationBlock = ^(RLMMigration *migration, uint64_t) {
+    RLMRealmConfiguration *config1 = self.config;
```

Comment:
This test passes without the newly added code, and is unrelated to the reported bug. The thing which didn't work is the following:

```
- (void)testEnumerateObjectTypeRemovedFromSchema {
    [self createTestRealmWithClasses:@[IntObject.class] block:^(RLMRealm *realm) {
        [IntObject createInRealm:realm withValue:@[@1]];
        [IntObject createInRealm:realm withValue:@[@2]];
    }];

    RLMRealmConfiguration *config = self.config;
    config.objectClasses = @[StringObject.class];
    config.schemaVersion = 1;
    __block int enumerateCalls = 0;
    config.migrationBlock = ^(RLMMigration *migration, uint64_t) {
        [migration enumerateObjects:IntObject.className block:^(RLMObject *oldObject, RLMObject *newObject) {
            XCTAssertNotNil(oldObject);
            XCTAssertNil(newObject);
            ++enumerateCalls;
            XCTAssertGreaterThan([oldObject[@"intCol"] intValue], 0);
        }];
    };
    XCTAssertTrue([RLMRealm performMigrationForConfiguration:config error:nil]);
    XCTAssertEqual(enumerateCalls, 2);
}
```